### PR TITLE
Fix ARM rust nightly issue in pathfinder_simd Stop using old-style simd_shuffle

### PR DESCRIPTION
**Pull request by spital**
Fixes ARM to compile, but the test on ARM fails when multithreading enabled (default).
Tested on f3c604acc Fri Sep 8 13:58:10 2023 +0200
Tests and builds fine on x64.
```
on NanoPC-T6 ARM worked as:
~/darkfi$ time cargo test -p darkfid2 --bin darkfid2 --release -- --test-threads=1
    Finished release [optimized] target(s) in 0.60s
     Running unittests src/main.rs (target/release/deps/darkfid2-f9b8cc71e489e83d)

running 2 tests
test tests::forks::forks ... ok
test tests::sync_blocks ... 
...
10:38:12 [INFO] [WASM] Contract log: [ConsensusStakeV1] Adding new coin to the set
10:38:12 [INFO] [WASM] Contract log: [ConsensusStakeV1] Adding new coin to the Merkle tree
10:38:12 [INFO] Gas used: 34495910/400000000
10:38:12 [INFO] Instantiating a new runtime
10:38:12 [INFO] Gas used: 6017741/400000000
10:38:12 [INFO] Gas used: 36907718/400000000
10:38:12 [INFO] [WASM] Contract log: [MintV1] Adding new coin to the set
10:38:12 [INFO] [WASM] Contract log: [MintV1] Adding new coin to the Merkle tree
10:38:12 [INFO] Gas used: 36936663/400000000
ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 496.85s


real    8m17.862s
user    30m14.499s
sys     0m11.044s

===
but with multithreading run fails

time cargo test -p darkfid2 --bin darkfid2 --release
...
16:42:40 [INFO] [WASM] Contract log: [ConsensusStakeV1] Adding new coin to the Merkle tree
16:42:40 [INFO] Gas used: 34458734/400000000
16:42:40 [INFO] Instantiating a new runtime
16:42:40 [INFO] Gas used: 6017741/400000000
16:42:41 [INFO] Gas used: 36870542/400000000
16:42:41 [INFO] [WASM] Contract log: [MintV1] Adding new coin to the set
16:42:41 [INFO] [WASM] Contract log: [MintV1] Adding new coin to the Merkle tree
16:42:41 [INFO] Gas used: 36899487/400000000
test tests::sync_blocks ... FAILED

failures:

---- tests::sync_blocks stdout ----
thread 'tests::sync_blocks' panicked at bin/darkfid2/src/tests/harness.rs:139:9:
assertion `left == right` failed
  left: 3
 right: 2
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    tests::sync_blocks

test result: FAILED. 1 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 348.29s

error: test failed, to rerun pass `-p darkfid2 --bin darkfid2`
make: *** [Makefile:92: test] Error 101

real    52m54.383s

... other cases ...
...
test result: FAILED. 1 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4466.90s

error: test failed, to rerun pass `-p darkfid2 --bin darkfid2`

real    74m28.268s
user    270m23.963s
sys     0m12.634s

```

### CODE REVIEW

**Comment by aggstam** [comment]
Hey @spital ,

Platform specific patches should be added on their corresponding file, not in repor Cargo.toml.
You can see an example here: [riscv.Dockerfile](https://github.com/darkrenaissance/darkfi/blob/master/contrib/docker/riscv.Dockerfile#L51-L56)

**Comment by spital** [comment]
Unlike RiscV, ARM is not "is highly experimental" - it was working before, and we not need  "have to add some hack patches", just the latest from pathfinder_simd that fixes its build (not tests) on ARM/aarch64 / Mac M1 and M2.

**Comment by aggstam** [comment]
The example showcases how to add patches to the Cargo.toml that are platform specific.
It wasn't a comment regarding platforms stability...

Main Cargo.toml should always use latest crate releases and patches that relate to code execution, not platform building.
Since the latest crate release doesn't use that patch yet, it shouldn't go to main Cargo.toml.
Each platform should be handled in their corresponding docker builder, where platform specific patches can be added like in the provided example.

tldr: add the patch in the arm dockerfiles(as per example) until crate release contains it

**Comment by spital** [comment]
There is no rush ... Dockerfiles are working on x64 and aarch64 perfectly fine, no need to add there anything.
There are `darkfi/contrib/docker/Docker*Emulation` that I used, but they may not be up to date and the build is slower than on NanoPC-t6.
ARM users can use
https://github.com/darkrenaissance/darkfi/pull/218.patch
to patch sources while the change is not in pathfinder_simd crate.

**Comment by aggstam** [comment]
Agreed!
Will keep this PR open so people can quickly find it, until crate release resolve this, but will not merge.
Thanks again for all your contributions! :)

**Comment by spital** [comment]
works now, closing
```
pi@NanoPC-T6:~/darkfi$ # time cargo test -p darkfid2 --bin darkfid2 --release
...
...
09:07:29 [INFO] Gas used: 34556751/400000000
09:07:29 [INFO] Instantiating a new runtime
09:07:29 [INFO] Gas used: 6009043/400000000
09:07:30 [INFO] Gas used: 36896102/400000000
09:07:30 [INFO] [WASM] Contract log: [GenesisMintV1] Adding new coin to the set
09:07:30 [INFO] [WASM] Contract log: [GenesisMintV1] Adding new coin to the Merkle tree
09:07:30 [INFO] Gas used: 36921727/400000000
test tests::sync_pos_blocks ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1191.99s


real    28m6.745s
user    128m47.901s
sys     1m13.998s

pi@NanoPC-T6:~/darkfi$ git status | head -5
HEAD detached at 9c74b776f
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   Cargo.lock

```
